### PR TITLE
Add AutoEnvProvider to generate DATETIME and PID environment variables

### PR DESCRIPTION
**Pull request by isseis**
This pull request implements the automatic environment variable provider (`AutoEnvProvider`) for the runner, along with comprehensive tests and updates to the implementation plan documentation. The main changes include defining the provider interface and its implementation, adding a function type for time injection, and ensuring robust test coverage for various edge cases.

**Implementation of automatic environment variable provider:**

* Added new file `autoenv.go` with the definition and implementation of `AutoEnvProvider`, including the `Clock` function type for dependency injection, constants for environment variable keys, and methods to generate current datetime (in UTC, formatted as `YYYYMMDDHHMM.msec`) and process ID. (`internal/runner/environment/autoenv.go`)

**Testing and validation:**

* Added comprehensive tests in `autoenv_test.go` to verify correct generation of environment variables, formatting of the datetime value, correct PID, handling of edge cases (such as year boundaries and nanosecond truncation), UTC conversion, and default clock behavior. (`internal/runner/environment/autoenv_test.go`)

**Documentation updates:**

* Updated the implementation plan (`04_implementation_plan.md`) to mark the completion of tasks related to the definition of constants and types, interface definition, implementation, and test creation for `AutoEnvProvider`. (`docs/tasks/0028_automatic_env_vars/04_implementation_plan.md`) [[1]](diffhunk://#diff-9c6e28d156467e1b0cf5e147fde4826e8eae3f88fefdf74657fa52d605d32131L19-R19) [[2]](diffhunk://#diff-9c6e28d156467e1b0cf5e147fde4826e8eae3f88fefdf74657fa52d605d32131L28-R34) [[3]](diffhunk://#diff-9c6e28d156467e1b0cf5e147fde4826e8eae3f88fefdf74657fa52d605d32131L43-R43)

### CODE REVIEW

File: internal/runner/environment/autoenv.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		AutoEnvPrefix + AutoEnvKeyPID:      p.generatePID(),
+	}
+}
+
+// generateDateTime generates the datetime string in YYYYMMDDHHMM.msec format (UTC)
+func (p *autoEnvProvider) generateDateTime() string {
+	now := p.clock().UTC()
+	dateTimePart := now.Format(DatetimeLayout)
+	msec := now.Nanosecond() / nanosToMillis
+	return fmt.Sprintf("%s.%03d", dateTimePart, msec)
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/environment/autoenv.go:70]: done

File: internal/runner/environment/autoenv_test.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	t.Run("DATETIME has correct format", func(t *testing.T) {
+		provider := NewAutoEnvProvider(nil)
+		result := provider.Generate()
+
+		datetime := result[AutoEnvPrefix+AutoEnvKeyDatetime]
+		// Format: YYYYMMDDHHMM.msec (e.g., "202510051430.123")
+		pattern := `^\d{12}\.\d{3}$`
+		matched, err := regexp.MatchString(pattern, datetime)
+		require.NoError(t, err)
+		assert.True(t, matched, "DATETIME should match pattern YYYYMMDDHHMM.msec, got: %s", datetime)
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/environment/autoenv_test.go]: done
